### ISSUE TITLE
Cleaning up Old Links and References in Ansible Documentation

### ISSUE THREAD

**Issue by iamgini** [ansible/ansible#74807]
### Summary

Dear Team,

I would like to start some cleaning up on Ansible Documentation (https://docs.ansible.com/) where we have so many links pointing to unmaintained versions/pages of documentations. 

Eg:
Page: https://docs.ansible.com/ansible/latest/plugins/connection.html
"paramiko SSH " link is pointing to an unmaintained version link (https://docs.ansible.com/ansible/2.8/plugins/connection/paramiko_ssh.html#paramiko-ssh-connection), instead of the latest collection information. 

1. Could anyone guide me that if this is okay to add entry there with latest link ?
2. Could you share some guideline for adding link to different page instead of simple :ref:`paramiko SSH<paramiko_ssh_connection>`

Thank you.

### Issue Type

Documentation Report

### Component Name

ansible/docs/docsite/rst/plugins/connection.rst 

### Ansible Version

```console
$ ansible --version
2.9
```

### Configuration

```console
$ ansible-config dump --only-changed
NA
```

### OS / Environment

All OS

### Additional Information

It would be great to link to the latest doc even from old pages. Otherwise, users may refer click and browse through old page references. 

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
* [`docs/docsite/rst/plugins/connection.rst`](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/plugins/connection.rst)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by felixfontein** [comment]
The source file for this specific issue is https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/plugins/connection.rst. This specific issue is caused because the wrong reference target is used: I think it should be `paramiko_connection`, not `paramiko_ssh_connection`. This is probably caused by the strange issue (for this specific connection plugin) that the file is `paramiko_ssh.py` (https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/connection/paramiko_ssh.py), while the plugin's name is `paramiko` (https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/connection/paramiko_ssh.py#L9). Adjusting `<paramiko_ssh_connection>` to `<paramiko_connection>` should fix it.

(The next link points to /ansible/3/ instead of /ansible/latest/, I don't know why. The link after that correctly points to /ansible/latest/.)

**Comment by felixfontein** [comment]
CC @acozine @samccann @lmalivert

**Comment by ginigangadharan** [comment]
@felixfontein thank you.
The latest should be `/latest/collections/ansible/builtin/paramiko_ssh_connection.html` right ?
so `paramiko_ssh_connection` supposed to be okay, but not working !!! :)

**Comment by felixfontein** [comment]
@ginigangadharan don't confuse filenames and RST labels. The filename/path is `/latest/collections/ansible/builtin/paramiko_ssh_connection.html`, but the RST label is `paramiko_connection`, and not `paramiko_ssh_connection`. (That's coming from https://github.com/ansible-community/antsibull/blob/main/antsibull/data/docsite/plugin.rst.j2#L19 and the fact that the `name:` entry in this plugin is historically 'wrong'.)

**Comment by ginigangadharan** [comment]
@felixfontein aah, I was confused. thank you and noted on that.

**Comment by samccann** [comment]
Hi @ginigangadharan - checking in to see if you are still planning on making these corrections?  Thanks!

**Comment by ginigangadharan** [comment]
HI @samccann,
Yes, I would like to proceed with clean up (and other updates if any). 
I was referring to the below pages for guidelines,
- [Contributing to the Ansible Documentation](https://docs.ansible.com/ansible/latest/community/documentation_contributions.html)
- [Docs Working Group](https://github.com/ansible/community/wiki/Docs)

I Hope, I am referring to the right documents :)

**Comment by samccann** [comment]
@ginigangadharan That would be great, thanks! Let me know if you have any questions etc.

**Comment by ginigangadharan** [comment]
Thank you @samccann.

**Comment by EstherChristopher** [comment]
Hi @ginigangadharan, are there other pages with similar error links that needs correction? I'll like to join you on this issue. Or how are you able to spot them?

**Comment by ginigangadharan** [comment]
@EstherChristopher 
There are but I didnt track since then.
I am planning to spend some time next month (hopefully) 

Thanks for asking.

**Comment by EstherChristopher** [comment]
@ginigangadharan 
Okay, if you do you can hmu. I'd like to do this too.

**Comment by ginigangadharan** [comment]
Sure. Thanks


*Gineesh*
linkedin.com/in/gineesh <https://www.linkedin.com/in/gineesh/>

**Comment by samccann** [comment]
@EstherChristopher feel free to fix the one identified issue [based on this comment](https://github.com/ansible/ansible/issues/74807#issuecomment-847157815).  And when @ginigangadharan has time, we can use this issue to identify any further problems they find.

Thanks to both of you!

**Comment by danush754** [comment]
Hlo @samccann, 

I'd want to go through the Ansible documentation and replace any broken links.

**Comment by samccann** [comment]
To be clear - this is a very broad issue and I'm sure there are multiple places in the docset that have link problems. So for any visitors to this issue, please just open a PR to fix the links you've found that need updating, and just mention this issue in your description.

And thanks everyone for the help!

**Comment by bcoca** [comment]
The simplest way might be to add a link checker to ansibullbot docs build, make them warnings for now but once removed they can become errors and be resolved before we publish them.

As for existing previous site versions a separate periodic check would be needed

**Comment by chaitu-0-7** [comment]
Hi @ginigangadharan , are there other pages that need correction? I'd like to join you on the issue !

**Comment by chaitu-0-7** [comment]
Hi @ginigangadharan are there still any document issues that needs to be cleared. I'd like to work on these.

**Comment by JaroslavKlech** [comment]
> The simplest way might be to add a link checker to ansibullbot docs build, make them warnings for now but once removed they can become errors and be resolved before we publish them.
> 
> As for existing previous site versions a separate periodic check would be needed

Was this already done?

**Comment by samccann** [comment]
@JaroslavKlech no, we don't have any link checkers on the ansible docs build yet. Last time I experimented, it got a lot of false positives (links listed as broken but weren't). But that was a couple of years ago.

**Comment by Manikkame123** [comment]
Hi Guys, I am beginner to this repo. Is this good issue to pickup as begginer ?

**Comment by samccann** [comment]
Hi @Manikkame123  - I just  took the easyfix label off this one. Sorry for the confusion. We're at the point now where we should find some kind of linkchecker to pull out a list of broken links before we can get folks on board to help fix them all.

**Comment by samccann** [comment]
closing this as we fixed the ones identified.
